language/bytecode_verifier/src/abstract_state.rs

```diff
@@ -575,7 +555,7 @@ impl AbstractState {
 
 impl AbstractDomain for AbstractState {
     /// attempts to join state to self and returns the result
-    /// both self.is_canonical() and state.is_canonical() must be true
+    /// both self and state must be canonical
```

Comment:
Let us keep is_canonical and add MIRAI preconditions asserting the statement in the documentation.  

It is fine to remove the other two if they are not being used.

Reply:
OK. I made that change and re-ran the coverage tests to confirm that the is_canonical function now shows up as being exercised in the tests.